### ISSUE TITLE
how to apply RSS transform for test data.. ? 

### ISSUE THREAD

**Issue by ajugeorge93** [facebookresearch/fastMRI#10]
RSS function is not returning tensor with all slices.

**Comment by anuroopsriram** [comment]
Can you please clarify your question?

One thing to note when using the `root_sum_of_squares` function is that it takes a `dim`ension as its second argument.  If you apply the `root_sum_of_squares` function to a single slice of shape `(num_coils, h, w)`, you should use `dim=0`, which is the default value. If you apply it on an entire volume of shape `(num_slices, num_coils, h, w)`, then you should use `dim=1`.

Also, if you want to apply RSS to complex inputs, make sure you apply `complex_abs` to compute absolute values before applying RSS.

**Comment by ajugeorge93** [comment]
I m trying to reconstruct an image from the test dataset 
the code I used is given below

`import h5py
import matplotlib.pyplot as  plt
filename = 'file1000022.h5'
import torch
f = h5py.File(filename, 'r')

# List all groups
print("Keys: %s" % f.keys())
a_group_key = list(f.keys())[0]
import numpy as np
# Get the data
data = np.array(f['kspace'])
mask=np.array(f['mask'])

from transforms import  *


tensor_data=to_tensor(data)
#print(tensor_data.shape)#torch.Size([37, 640, 368, 2])
fft_value=fft2(tensor_data)

complex_value=complex_abs(fft_value)
print(complex_value.shape)#torch.Size([37, 640, 368])
root_sum_of_squares=root_sum_of_squares(complex_value,dim=1)
print(root_sum_of_squares.shape) # torch.Size([37, 368])`

but return a tensor of shape [37,368] expected is [37,640,368] why RSS is not applied for all slices. I am a beginner in MRI reconstruction. please pardon my mistakes

**Comment by anuroopsriram** [comment]
Is this the single-coil data? For the single-coil data, root sum of squares does not make sense.
For multi-coil data, `print(tensor_data.shape)` should return something like `torch.Size([37, 15, 640, 368, 2])`.

**Comment by ajugeorge93** [comment]
yes it is single coil data .. could you help with the steps and functions used to reconstruct single coil and multicoil. It will be useful for beginners like me.

**Comment by anuroopsriram** [comment]
You can look at the example models to see what we do: [run_unet.py](https://github.com/facebookresearch/fastMRI/blob/master/models/unet/run_unet.py) and [run_bart_test.py](https://github.com/facebookresearch/fastMRI/blob/master/models/cs/run_bart_test.py).

In the multi-coil case, you can use the `root_sum_of_squares` function to combine the different coils. It is not needed for the single-coil case. See [run_unet.py:73](https://github.com/facebookresearch/fastMRI/blob/master/models/unet/run_unet.py#L73) for example.

**Comment by ajugeorge93** [comment]
I have reconstructed an image from single coil Test data but the reconstructed image seems like more noise.
The output Image of slice number 16 [https://drive.google.com/file/d/11icZEzwS2jf_swoLBrHTB7STcHGQFa32/view?usp=sharing](here). How to improve the reconstructed image with reduced noise.

thanks in advance 

```import h5py
import matplotlib.pyplot as  plt
filename = 'file1000036.h5'
import torch
f = h5py.File(filename, 'r')

# List all groups
print("Keys: %s" % f.keys())
a_group_key = list(f.keys())[0]
import numpy as np
# Get the data
kspace = np.array(f['kspace'])
mask=np.array(f['mask'])

from models.unet import train_unet
from common import subsample

center_fractions=[0.08, 0.04]
accelerations=[4, 8]
mask_func=subsample.MaskFunc(center_fractions, accelerations)
resolution=321
which_challenge='singlecoil'
import transforms

kspace = transforms.to_tensor(kspace)
# Apply mask
use_seed=True
if mask_func is not None:
    seed = tuple(map(ord, filename))
    masked_kspace, _ = transforms.apply_mask(kspace, mask_func, seed)
else:
    masked_kspace = kspace

# Inverse Fourier Transform to get zero filled solution
image = transforms.ifft2(masked_kspace)
# Crop input image
image = transforms.complex_center_crop(image, (resolution, resolution))
# Absolute value
image = transforms.complex_abs(image)
# Apply Root-Sum-of-Squares if multicoil data
if which_challenge == 'multicoil':
    image = transforms.root_sum_of_squares(image)
# Normalize input
image, mean, std = transforms.normalize_instance(image, eps=1e-11)
image = image.clamp(-6, 6)

print(image.shape)
plt.imshow(image[16],cmap='gray')
plt.show()```

**Comment by anuroopsriram** [comment]
That is because you are using the test data, which is already sub-sampled. Can you try the same with training data?

**Comment by ajugeorge93** [comment]
Thank you. Now I understood.
